File: Core/Entities/Product.cs
namespace Core.Entities
{
    public partial class Product
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public string Photo { get; set; }

        public int ProductTypeId { get; set; }
        public virtual ProductType ProductType { get; set; }

        public int ProductBrandId { get; set; }
        public virtual ProductBrand ProductBrand { get; set; }
    }
}

File: UI/barbershop_mobile/lib/models/product.dart
import 'package:json_annotation/json_annotation.dart';

part 'product.g.dart';

@JsonSerializable()
class Product {
  int? id;
  String? name;
  String? description;
  double? price;
  String? pictureUrl;
  String? productType;
  String? productBrand;
  String? photo;

  Product(
      {this.id,
      this.name,
      this.description,
      this.price,
      this.pictureUrl,
      this.productType,
      this.productBrand,
      this.photo});

  factory Product.fromJson(Map<String, dynamic> json) =>
      _$ProductFromJson(json);

  Map<String, dynamic> toJson() => _$ProductToJson(this);
}
